lib/Agrammon/DB/Dataset.pm6

```diff
@@ -542,4 +507,79 @@ class Agrammon::DB::Dataset does Agrammon::DB {
         }
     }
 
+    method store-branch-data(@vars, Str $instance, %options, @fractions) {
+        my $dataset-name = $!name;
+
+        # pg-array syntax: "{1,2,3}"
+        my $fractions-pg = '{' ~ @fractions[*;*].join(',') ~ '}';
+
+        my @branch-variables;
+        # Get variable ids and names
+        self.with-db: -> $db {
+            my $username = $!user.username;
+            @branch-variables = $db.query(q:to/SQL/, $username, $dataset-name, |@vars, $instance).hashes;
+            SELECT data_id, data_var
+                  FROM data_new
+                 WHERE data_dataset=dataset_name2id($1,$2)
+                   AND data_var IN ($3,$4)
+                   AND data_instance = $5
+                   ORDER BY data_id
+            SQL
+        }
+
+        for @branch-variables -> %var {
+            my $var-id   = %var<data_id>;
+            my $var-name = %var<data_var>;
+
+            # pg array syntax: { "test_1", "test_2" }
+            my $options-pg = '{' ~ %options{$var-name}.map(*.subst(' ', '_', :g) ).join(',') ~ '}';
+
```

Comment:
The `DB::Pg` module can handle Postgres arrays;, see the [short Arrays section in the docs](https://modules.raku.org/dist/DB::Pg:cpan:CTILMES). So you should be able to send in a Raku array and have it do the right thing.

Reply:
Oh, neat. That makes it 0 instead of 11 lines :-)